clippy_lints/src/swap.rs

```diff
@@ -172,15 +172,16 @@ fn check_manual_swap(cx: &LateContext<'_>, block: &Block<'_>) {
     }
 }
 
+#[allow(clippy::too_many_lines)]
 /// Implementation of the `ALMOST_SWAPPED` lint.
 fn check_suspicious_swap(cx: &LateContext<'_>, block: &Block<'_>) {
-    for w in block.stmts.windows(2) {
+    for stmt in block.stmts.windows(2) {
         if_chain! {
-            if let StmtKind::Semi(first) = w[0].kind;
-            if let StmtKind::Semi(second) = w[1].kind;
-            if first.span.ctxt() == second.span.ctxt();
-            if let ExprKind::Assign(lhs0, rhs0, _) = first.kind;
-            if let ExprKind::Assign(lhs1, rhs1, _) = second.kind;
+            if let StmtKind::Semi(first) = stmt[0].kind;
+            if let StmtKind::Semi(second) = stmt[1].kind;
+            if let Some((lhs0, rhs0)) = parse(&stmt[0]);
+            if let Some((lhs1, rhs1)) = parse(&stmt[1]);
```

Comment:
Here `parse` should be acting like https://github.com/rust-lang/rust-clippy/pull/8945#issuecomment-1192657937, where it checks for either a `let a = b` or `a = b`. That way we can get rid of https://github.com/rust-lang/rust-clippy/pull/10177/files#diff-342cbd825f201119c85fd8892ed86fb369d63b914c1e1e0179a40ef10215bd44R225-R295

```suggestion
            if let Some((lhs0, rhs0)) = parse(&stmt[0]);
            if let Some((lhs1, rhs1)) = parse(&stmt[1]);
```